File: packages/sdk-typescript/examples/quickstart.ts
import { HttpTransport, MachinaClient } from "@machina/sdk-typescript";

const client = new MachinaClient(new HttpTransport("http://127.0.0.1:8080"));
const session = await client.createSession();
try {
  await session.navigate("https://fixture.local/");
  const result = await session.page().extract("main article");
  console.log(result.result);
} finally {
  await session.close();
}

File: packages/sdk-python/examples/quickstart.py
import asyncio

from machina_sdk import HttpTransport, MachinaClient


async def main() -> None:
    client = MachinaClient(HttpTransport("http://127.0.0.1:8080"))
    session = await client.create_session()
    try:
        await session.navigate("https://fixture.local/")
        result = await session.page().extract("main article")
        print(result.result)
    finally:
        await session.close()


if __name__ == "__main__":
    asyncio.run(main())
